src/traffic_layout/engine.cc

```diff
@@ -454,6 +463,28 @@ RunrootEngine::copy_runroot(const std::string &original_root, const std::string
   original_map[LAYOUT_INFODIR]       = TS_BUILD_INFODIR;
   original_map[LAYOUT_CACHEDIR]      = TS_BUILD_CACHEDIR;
 
+  RunrootMapType new_map = original_map;
+  // use the user provided layout: layout_file
+  if (layout_file.size() != 0) {
+    try {
+      YAML::Node yamlfile = YAML::LoadFile(layout_file);
+      for (auto it : yamlfile) {
+        std::string key   = it.first.as<std::string>();
+        std::string value = it.second.as<std::string>();
+        if (new_map.find(key) != new_map.end()) {
```

Comment:
If you do a `find` on a map, keep the iterator for doing the assignment, otherwise you'd doing the lookup twice.